diff --git i/package.json w/package.json
index 5dc75e2a..59b69621 100644
--- i/package.json
+++ w/package.json
@@ -182,7 +182,7 @@
     "typescript-eslint": "^8.56.1"
   },
   "engines": {
-    "node": "^22.19.0",
+    "node": "^25.0.0",
     "pnpm": "^10.0.0"
   },
   "config": {
diff --git i/server/index.ts w/server/index.ts
index cd34d7d5..54fd47cf 100644
--- i/server/index.ts
+++ w/server/index.ts
@@ -180,7 +180,7 @@ app
       }
     });
     if (settings.network.csrfProtection) {
-      server.use(
+      server.use(() =>
         csurf({
           cookie: {
             httpOnly: true,
